src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentProjectsFinder.cs

```diff
@@ -364,6 +364,11 @@ private static Lazy<HashSet<string>> CreateInternalsVisibleToMap(IAssemblySymbol
                     }
 
                     var value = (string)typeNameConstant.Value;
+                    if (string.IsNullOrWhiteSpace(value))
+                    {
+                        continue;
+                    }
+
                     var commaIndex = value.IndexOf(',');
                     var assemblyName = commaIndex >= 0 ? value.Substring(0, commaIndex).Trim() : value;
```

Comment:
Yes, I also thought of adding further validation here to check for obviously invalid assembly names (empty, all whitespace, illegal chars, etc.). However, in the spirit of fixing the Watson bug, I left it alone. On that note, one could argue that "IsNullOrWhitespace" is more than required to fix the NRE - a simple null check would be sufficient.